lib/modules/dosomething/dosomething_reportback/dosomething_reportback.module

```diff
@@ -439,8 +406,30 @@ function dosomething_reportback_admin_view_entity($entity) {
 /**
  * Callback for /rbf/FID page.
  */
-function dosomething_reportback_file_view_entity($entity) {
-  return entity_view('reportback_file', array($entity->rbid => $entity), 'full');
+function dosomething_reportback_view_entity($entity) {
+  $rb_user = dosomething_user_load(user_load($entity->uid));
+  global $user;
+  // For now, only display this page to the user, unless it's a staff member.
+  if (($is_current = $user->uid == $rb_user->uid) || dosomething_user_is_staff($user)) {
+    $fid = $_GET['fid'];
+    if ($fid) {
+      $rb_image = dosomething_image_get_themed_image_by_fid($fid, '300x300');
+    }
+    else {
+      $rb_image = dosomething_image_get_themed_image_by_fid($entity->fids[0], '300x300');
+    }
+    $node = dosomething_campaign_load(node_load($entity->nid));
+
+    return theme('reportback_permalink', array(
+      'rb_image' => $rb_image,
+      'reportback' => $entity,
+      'node' => $node,
+      'user' => $rb_user,
+      'is_current' => $is_current,
+      )
+    );
+  }
+  return drupal_not_found();
```

Comment:
Does this act weird? Usually it seems to just to be called without the `return` , e.g. https://api.drupal.org/api/drupal/includes%21file.inc/function/file_download/7